### PR TITLE
chore: bulk update and fix dependencies [NONE]

### PR DESCRIPTION
**Pull request by marcolink**
Bulk update several dependencies and lock `semantic-release` to v19 as any higher major version requires node@18

### CODE REVIEW

**Review by crissto — APPROVED** [review]
Nice! 🚢

File: lib/constants/editor-interface-defaults/types.ts [hunk 1]
```diff
@@ -11,4 +11,5 @@ export const DEFAULT_EDITOR_ID = 'default-editor'
 /**
  * @private
  */
-export const in_ = <K extends string, O>(key: K, object: O): key is K & keyof O => key in object
+export const in_ = <K extends string, O extends object>(key: K, object: O): key is K & keyof O =>
```
Comment (andreascful) [inline lib/constants/editor-interface-defaults/types.ts:14]: What exactly does this change do?

Comment (crissto) [reply lib/constants/editor-interface-defaults/types.ts:14]: It forces O to be a type of object, so you can't pass things like string or number where keyof wouldn't give you the wanted result

Comment (marcolink) [reply lib/constants/editor-interface-defaults/types.ts:14]: What @crissto says. The change was required by more strict rules of the latest Typescript version

**Pull request merged**